src/averageAges.js

```diff
@@ -54,8 +84,28 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  const mothers = people.filter((person) => person.sex === 'f');
+  let children;
+
+  if (onlyWithSon) {
+    children = people.filter((person) => person.sex === 'm');
+  } else {
+    children = people.filter((person) => person.mother !== undefined);
+  }
+
+  const diffs = children.map((child) => {
+    const mother = mothers.find((person) => person.name === child.mother);
```

Comment:
you may use `people` array here, i.e `mothers` variable is redundant (you are checking `mother` field, so there won't be match with people that have `sex === 'm'`)